cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,251 @@
+                        base_section = self.base_config.get_section(section.key, section.label)
+                        base_value = base_section.get_param(param.key).value if base_section else None
+                        target_value = param.value
+
+                        if base_value != target_value:
+                            self.changes.append(
+                                Change(
+                                    section.key,
+                                    section.label,
+                                    param.key,
+                                    base_value,
+                                    param.value,
+                                    param.get_updatability(),
+                                )
+                            )
+
+    def _restore_settings_params(self, config):
+        cluster_section = config.get_section("cluster")
+        for _, param in cluster_section.params.items():
+            if param.key.endswith("_settings"):
+                self._restore_settings_param(config, param)
+
+    def _restore_settings_param(self, config, settings_param):
+        sections_labels_list = []
+        sections = config.get_sections(settings_param.referred_section_key).items()
+
+        for _, section in sections:
+            if not hasattr(section, "mock"):
+                sections_labels_list.append(section.label)
+        settings_param.value = ",".join(sorted(sections_labels_list))
+
+    def _rename_base_sections(self):
+        """
+        Rename sections in base config.
+
+        Make all sections labels in base config different from labels in target config, so that after adaptation we can
+        check that all sections have been correctly matched by just comparing their labels.
+        """
+        for section_key in self.base_config.get_section_keys():
+            new_lbl = u"_{0}{1}"
+            i = 0
+            for _, base_section in sorted(self.base_config.get_sections(section_key).items()):
+                base_section.label = new_lbl.format(section_key, i if i > 0 else "")
+                i += 1
+
+    @property
+    def updatability_level(self):
+        """Get the updatability of the ConfigPatch."""
+        return max(change.updatability.level for change in self.changes) if len(self.changes) else Updl.ALLOWED
+
+    def _create_default_section(self, config, section):
+        """Create a default section of same type of the provided section in the provided config to allow comparison."""
+        section_definition = section.definition
+        section_type = section_definition.get("type")
+        mock_section = section_type(
+            section_definition=section_definition, pcluster_config=self.target_config, section_label=section.label
+        )
+        # Sections with key param (like EBS) are marked as mock to be ignored from related settings params
+        if section.key_param:
+            mock_section.mock = True
```

Comment:
I'd prefer keeping the mock attribute (or any other explicit way) to avoid unexpected behaviours like removing unexpected sections and to make explicit the purpose of the code.